crates/ra_ide/src/ssr.rs

```diff
@@ -225,18 +244,28 @@ fn find(pattern: &SsrPattern, code: &SyntaxNode) -> SsrMatches {
 fn replace(matches: &SsrMatches, template: &SsrTemplate) -> TextEdit {
     let mut builder = TextEditBuilder::default();
     for match_ in &matches.matches {
-        builder.replace(match_.place.text_range(), render_replace(&match_.binding, template));
+        builder.replace(
+            match_.place.text_range(),
+            render_replace(&match_.binding, &match_.ignored_comments, template),
+        );
     }
     builder.finish()
 }
 
-fn render_replace(binding: &Binding, template: &SsrTemplate) -> String {
+fn render_replace(
+    binding: &Binding,
+    ignored_comments: &Vec<SyntaxToken>,
+    template: &SsrTemplate,
+) -> String {
     let mut builder = TextEditBuilder::default();
     for element in template.template.descendants() {
         if let Some(var) = template.placeholders.get(&element) {
             builder.replace(element.text_range(), binding[var].to_string())
         }
     }
+    for comment in ignored_comments {
+        builder.insert(template.template.text_range().end(), comment.to_string())
```

Comment:
Yeah, not sure what's the correct way to handle comments here....

I would expect that we might want to tweak comment attacmhent rules, such that comments "on" elements are placed inside the element's node in the syntax tree. Trivia attachment is handled here:

https://github.com/rust-analyzer/rust-analyzer/blob/1fe48a0115c24240f5a3e1b329e642f18e2715d6/crates/ra_syntax/src/parsing/text_tree_sink.rs#L144-L169

Reply:
Ideally, we'd put comments after whatever token in the replacement that matched the original token, right? :thinking:

Reply:
Like just put it immediately after whatever the last wildcard expression matched from the pattern was? I imagine that would probably do ok in most cases.